doc/modules/language-guide/pages/language-manual.adoc

```diff
@@ -1276,22 +1276,49 @@ The declaration `<dec>` of a `system` field must be a manifest `func` declaratio
 |  name | type | description
 
 | `heartbeat`  | `+() -> async ()+` | heartbeat action
+| `inspect_message` | `+{ caller : Principal; msg : <Variant>; arg : Blob } -> Bool+` | message predicate
 | `preupgrade`  | `+() -> ()+` | pre upgrade action
 | `postupgrade` | `+() -> ()+` | post upgrade action
+
 |===
 
 * `heartbeat`, when declared, is called on every Internet Computer subnet *heartbeat*,
 scheduling an asynchronous call to the `heartbeat` function. Due to its `async` return type, a heartbeat function may send messages and await results. The result of a heartbeat call, including
 any trap or thrown error, is ignored. The implicit context switch means that
 the time the heartbeat body is executed may be later than the time the
 heartbeat was issued by the subnet.
+* `inspect_message`, when declared, is called as a predicate on every Internet Computer ingress message (with the exception of queries). The return value, a `Bool`, indicates whether to accept or decline the given message. The argument type depends on the interface of the enclosing actor (see <<inspect-message>>).
```

Comment:
You told that queries going through concensus will also pass through `inspect_message`. Also we might explain that canister-to-canister sends are not filtered.

Reply:
```suggestion
* `inspect_message`, when declared, is called as a predicate on every {IC} ingress message (with the exception of queries). The return value, a `Bool`, indicates whether to accept or decline the given message. The argument type depends on the interface of the enclosing actor (see <<inspect-message>>).
```